### DISCUSSION TITLE
Asynchronous DMA transfers in Arduino_Bus

### DISCUSSION THREAD

**Discussion by calint** [moononournation/Arduino_GFX#473] [Ideas]
I have a draft of a patch that implements:

Arduino_Bus:
virtual bool asyncDMASupported();
virtual bool asyncDMAIsBusy();
virtual void asyncDMAWaitForCompletion();
virtual void asyncDMAWriteBytes(uint8_t *data, uint32_t len);

and implementation for:
Arduino_ESP32SPI
Arduino_ESP32SPIDMA

tested on ESP32-2432S028R with doubled performance in use case.

Kind regards

**Comment by calint** [comment]
and now implementation for:
Arduino_ESP32QSPI

tested on JC4827W543R with roughly doubled performance in use case.

Kind regards

**Comment by calint** [comment]
Pull request https://github.com/moononournation/Arduino_GFX/pull/475

**Comment by h5n1xp** [comment]
If @moononournation could include these in the master, I think I can implement DMA for the Raspberry Pi Pico (PR2040 and RP2350).
